### PR TITLE
dtl: support gcc 14

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **dtl/***

#### Motivation
To support gcc 14, make `enableTrivial` mutable.
https://github.com/cubicdaiya/dtl/pull/18

#### Details

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

File: recipes/dtl/all/conanfile.py [hunk 1]
```diff
@@ -18,7 +17,6 @@ class DtlConan(ConanFile):
 
     package_type = "header-library"
     settings = "os", "arch", "compiler", "build_type"
-    no_copy_source = True
```
Comment (AbrilRBS) [inline recipes/dtl/all/conanfile.py:21]: @toge we can keep this if we move the replace_in_file to the source() method, as every configuration will need it, and we save some space :)

**Review by AbrilRBS — COMMENTED** [review]
Approving this as an executive decision, this will not impact the generated code, it was wrong before! If removing const, at most we would lose on optimizations, but that's something good. Waiting on @toge to address my suggestion before merging :)

File: recipes/dtl/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/dtl/all/conanfile.py]: Thank you for your review!
I agree with you. Fixed.

**Review by perseoGI — APPROVED** [review]
Approving as an exception

**Review by AbrilRBS — APPROVED** [review]

**Pull request merged**